controllers/provider-packet/pkg/webhook/controlplanebackup/ensurer.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane/genericmutator"
+
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewEnsurer creates a new controlplaneexposure ensurer.
+func NewEnsurer(etcdBackup *config.ETCDBackup, imageVector imagevector.ImageVector, logger logr.Logger) genericmutator.Ensurer {
+	return &ensurer{
+		etcdBackup:  etcdBackup,
+		imageVector: imageVector,
+		logger:      logger.WithName("packet-controlplanebackup-ensurer"),
+	}
+}
+
+type ensurer struct {
+	genericmutator.NoopEnsurer
+	etcdBackup  *config.ETCDBackup
+	imageVector imagevector.ImageVector
+	client      client.Client
+	logger      logr.Logger
+}
+
+// InjectClient injects the given client into the ensurer.
+func (e *ensurer) InjectClient(client client.Client) error {
+	e.client = client
+	return nil
+}
+
+// EnsureETCDStatefulSet ensures that the etcd stateful sets conform to the provider requirements.
+func (e *ensurer) EnsureETCDStatefulSet(ctx context.Context, ss *appsv1.StatefulSet, cluster *extensionscontroller.Cluster) error {
+	if err := e.ensureContainers(&ss.Spec.Template.Spec, ss.Name, cluster); err != nil {
+		return err
+	}
+	return e.ensureChecksumAnnotations(ctx, &ss.Spec.Template, ss.Namespace, ss.Name)
+}
+
+func (e *ensurer) ensureContainers(ps *corev1.PodSpec, name string, cluster *extensionscontroller.Cluster) error {
+	c, err := e.getBackupRestoreContainer(name, cluster)
+	if err != nil {
+		return err
+	}
+	ps.Containers = controlplane.EnsureContainerWithName(ps.Containers, *c)
+	return nil
+}
+
+func (e *ensurer) ensureChecksumAnnotations(ctx context.Context, template *corev1.PodTemplateSpec, namespace, name string) error {
+	if name == common.EtcdMainStatefulSetName {
+		return controlplane.EnsureSecretChecksumAnnotation(ctx, template, e.client, namespace, packet.BackupSecretName)
+	}
+	return nil
+}
+
+func (e *ensurer) getBackupRestoreContainer(name string, cluster *extensionscontroller.Cluster) (*corev1.Container, error) {
+	// Find etcd-backup-restore image
+	// TODO Get seed version from clientset when it's possible to inject it
+	image, err := e.imageVector.FindImage(packet.ETCDBackupRestoreImageName, "", cluster.Shoot.Spec.Kubernetes.Version)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not find image %s", packet.ETCDBackupRestoreImageName)
+	}
+
+	const (
+		defaultSchedule = "0 */24 * * *"
+	)
+
+	// Determine provider and container env variables
+	// They are only specified for the etcd-main stateful set (backup is enabled)
+	var (
+		provider                string
+		env                     []corev1.EnvVar
+		volumeClaimTemplateName = name
+	)
+	if name == common.EtcdMainStatefulSetName {
+		env = []corev1.EnvVar{
```

Comment:
See above - the etcd-backup-restore sidecar doesn't have support for Packet - let's remove the env variables.